aeneas/src/mach/MachLowering.v3

```diff
@@ -1862,6 +1891,38 @@ class MachLowering(mach: MachProgram, compiler: Compiler, config: MachLoweringCo
 		}
 		return loads;
 	}
+	def genNormTypedLoadsByteRegion(source: Source, machType: Type, base: SsaInstr, offset: int) -> Array<SsaInstr> {
+		var tn: TypeNorm;
+		match (machType) {
+			x: IntType => tn = mach.intNorm.makeType(x);
+			_ => return [ptrLoad(machType, base, offset)];
+		}
+
+		var loads = Array<SsaInstr>.new(tn.size);
+
+		for (i < loads.length) {
```

Comment:
As discussed, I think you only need to worry about handling partial reads for the last `tn.sub[i]`. You can over-read for a field that is not too close to the end of a struct. We can do that optimization later and control it with a `Fact` set on the instruction.